test/unit/auth/authenticate-jwt-token.js

```diff
@@ -0,0 +1,290 @@
+'use strict';
+
+const path = require('path');
+const httpMocks = require('node-mocks-http'),
+      jwt = require('jsonwebtoken'),
+      should = require('should'),
+      sinon = require('sinon');
+
+
+const models = require(path.resolve('./models'));
+const authController = require(path.resolve('./controllers/authenticate-token'));
+const jwtConfig = require(path.resolve('./config/secret/jwt-config'));
+
+
+
+describe.only('generateTokenBehavior(req)', function() {
```

Comment:
I was/am experimenting with the best unit testing techniques for middlewares, and here I was separating behavior of generateToken middleware from actual middleware. Just here because I found it a little too confusing to continue with this practice in more places.